rs/bitcoin/kyt/src/providers.rs

```diff
@@ -3,15 +3,92 @@ use ic_btc_interface::Txid;
 use ic_cdk::api::management_canister::http_request::{
     CanisterHttpRequestArgument, HttpHeader, HttpMethod, TransformContext, TransformFunc,
 };
+use std::cell::RefCell;
+use std::fmt;
 
-pub fn create_request(
+/// Return the next bitcoin API provider for the given `btc_network`.
+///
+/// Internally it remembers the previously used provider in a thread local
+/// state and would iterate through all providers in a round-robin manner.
+pub fn next_provider(btc_network: BtcNetwork) -> Provider {
+    PREVIOUS_PROVIDER_ID.with(|previous| {
+        let provider = (Provider {
+            btc_network,
+            provider_id: *previous.borrow(),
+        })
+        .next();
+        *previous.borrow_mut() = provider.provider_id;
+        provider
+    })
+}
+
+thread_local! {
+    static PREVIOUS_PROVIDER_ID: RefCell<ProviderId> = const { RefCell::new(ProviderId::BtcScan) };
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+enum ProviderId {
+    BtcScan,
+    BlockStream,
```

Comment:
Nit: How about using the spelling `Btcscan` and `Blockstream`, which is how they spell it themselves?
`MempoolSpace` is different because `.space` is the domain (and I agree, it would be weird to just call it `Mempool`).